### PR TITLE
implement anvil and enchantment table inventorys

### PR DESCRIPTION
**Pull request by cooltexture1**
The inventorys are working at this stage but some of the code especially for the enchantment table is a mess. I will clean it up soon. If there are problems in the way i implemented these or bugs, let me know. This is some progress to #2

### CODE REVIEW

**Comment by terrarier2111** [comment]
Is this still in work or did you just forget to un-draft it?

**Comment by cooltexture1** [comment]
yes, now it should be ready for review

File: src/inventory/mod.rs [hunk 1]
```diff
@@ -472,9 +489,10 @@ impl Item {
     }
 }
 
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
 pub enum Material {
-    Air,                             // 1.7.10 (id: 0, stack: 0)| 1.13 (id: 9648)
+    #[default]
+    Air, // 1.7.10 (id: 0, stack: 0)| 1.13 (id: 9648)
```
Comment (terrarier2111) [inline src/inventory/mod.rs:495]: could you shift this comment back?

File: src/inventory/enchanting_table_inventory.rs [hunk 2]
```diff
@@ -0,0 +1,556 @@
...
+                enchantment_level: option_value,
+                ..this.button_data[i]
+            }
+        };
+
+        match property {
+            0 => update_level_required(0, self),
+            1 => update_level_required(1, self),
+            2 => update_level_required(2, self),
+            3 => (),
```
Comment (terrarier2111) [inline src/inventory/enchanting_table_inventory.rs:107]: Could you add a comment explaining why this has no action associated with it?

File: src/inventory/enchanting_table_inventory.rs [hunk 3]
```diff
@@ -0,0 +1,556 @@
...
+            4 => update_enchantment_hint(0, self),
+            5 => update_enchantment_hint(1, self),
+            6 => update_enchantment_hint(2, self),
+            7 => update_enchantment_level(0, self),
+            8 => update_enchantment_level(1, self),
+            9 => update_enchantment_level(2, self),
+            _ => warn!("the server sent invalid data for the enchanting table"),
+        }
+
+        // this fixes a bug in mc protocol
```
Comment (terrarier2111) [inline src/inventory/enchanting_table_inventory.rs:117]: this is a bit vague, could you elaborate on that?

**Review by terrarier2111 — CHANGES_REQUESTED** [review]

**Comment by terrarier2111** [comment]
This is really cool, thanks!

File: src/inventory/mod.rs (hunk 1 above)
Comment (cooltexture1) [reply src/inventory/mod.rs:495]: I tried doing this already, but for some reason `cargo fmt` shifts this one line forward every time it is run.

**Pull request merged**